yt_dlp/downloader/external.py

```diff
@@ -304,6 +319,130 @@ def _make_cmd(self, tmpfilename, info_dict):
             cmd += ['--', info_dict['url']]
         return cmd
 
+    def _call_downloader(self, tmpfilename, info_dict):
+        info_dict['__rpc_port'] = find_available_port() or 19190
+        info_dict['__rpc_secret'] = str(uuid.uuid4())
+        return super()._call_downloader(tmpfilename, info_dict)
+
+    def aria2c_rpc(self, rpc_port, rpc_secret, method, params, secret=True):
+        if secret and rpc_secret:
+            params = [f'token:{rpc_secret}', *params]
+        # note: there's no need to be UUID (it can even a numeric value), but that's easier
```

Comment:
```suggestion
        # note: there's no need to be UUID (it can even a numeric value), but that's easier and long enough not to make duplicate one
```